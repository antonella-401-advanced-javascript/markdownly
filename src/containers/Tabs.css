.Button {
  display: flex;
  flex-direction: row-reverse;

  button[name=add] {
    max-height: 35px;
    margin: 12px 0 2px 20px;
    padding: 2px 10px;
    font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
    border-radius: 35%;
    background-color: black;
    color: rgb(247, 238, 229);
    font-size: 12pt;
  }

  button[name=add]:hover {
    cursor: pointer;
    background-color: rgb(247, 238, 229);
    color: black;
    border: 1px solid rgba(0, 0, 0, 0.493);
  }
}