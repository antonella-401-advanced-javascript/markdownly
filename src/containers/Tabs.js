import React from 'react';
import TabDeck from '../components/tab/TabDeck';
import { useDispatch, useSelector } from 'react-redux';
import { getSearchTerm, getMatchTitlesByTitle } from '../selectors/documentSelectors';
import { addTab, deleteTab, addSearch, getMatchTitle } from '../actions/documentActions';
import Filter from '../components/filter/Filter';
import styles from './Tabs.css';

const Tabs = () => {
  const dispatch = useDispatch();
  const handleClick = tabNumber => dispatch(addTab(tabNumber));
  const handleDelete = title =>  dispatch(deleteTab(title));
  const handleSubmit = searchTerm => (event.preventDefault(), dispatch(getMatchTitle(searchTerm)));
  const handleChange = ({ target }) => (event.preventDefault(),
  dispatch(addSearch(target.value)));
  const titles = useSelector(getMatchTitlesByTitle);
  const searchTerm = useSelector(getSearchTerm);
  return (
    <div className={styles.Button}>
      <TabDeck handleDelete={handleDelete} titles={titles} />
      <button name="add" onClick={() => handleClick(titles.length)}>+</button>
      <Filter handleSubmit={handleSubmit} handleChange={handleChange} searchTerm={searchTerm} />
    </div>
  );
};

export default Tabs;