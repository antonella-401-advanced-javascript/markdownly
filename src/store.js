/* eslint-disable no-console */
import { createStore, applyMiddleware } from 'redux';
import reducer from './reducers/index';

const loggerMiddleware = store => next => action => {
  console.log('in my logger', action, store.getState());

  next(action);

  console.log('current state', store.getState());
};

export default createStore(
  reducer,
  applyMiddleware(
    loggerMiddleware
  )
);
