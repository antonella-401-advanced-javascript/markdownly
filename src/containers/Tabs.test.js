import React from 'react';
import { shallow } from 'enzyme';
import Tabs from './Tabs';
import store from '../store';

describe('Tabs component', () => {
  it('renders Tabs', () => {
    React.useContext = () => ({ store });
    const wrapper = shallow(<Tabs store={store}/>, {
      context: { store }
    });
    expect(wrapper).toMatchSnapshot();
  });
}); 
