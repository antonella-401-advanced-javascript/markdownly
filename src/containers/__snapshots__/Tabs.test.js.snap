// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders Tabs 1`] = `
<Fragment>
  <TabDeck
    handleDelete={[Function]}
    titles={
      Array [
        "default",
      ]
    }
  />
  <button
    onClick={[Function]}
  >
    ADD
  </button>
  <Filter
    handleChange={[Function]}
    handleSubmit={[Function]}
    searchTerm=""
  />
</Fragment>
`;
